homeassistant/components/trafikverket_train/coordinator.py

```diff
@@ -117,6 +124,18 @@ async def _async_update_data(self) -> TrainData:
                 f"Train departure {when} encountered a problem: {error}"
             ) from error
 
+        depart_next = None
+        depart_next_next = None
+        if not state and states and len(states) > 0:
+            state = states[0]
+            depart_next = states[1].advertised_time_at_location if states[1] else None
```

Comment:
Since when is `if states[1]` the same like `if key in states`?

Reply:
states is a list containing `TrainStop` objects so we only need to check if index 1 of that list has an object and thereby can pull the required info. Or am I not understanding what you mean?

Reply:
If states only has 1 entry, doesn't `states[1]` raise an IndexError?